### PR TITLE
[WIP]Fix schedule A zip search

### PR DESCRIPTION
**Pull request by hcaofec**
This PR will refactor the zip code search for SA. No matter what zip code a user enters, it will always find returns transactions  that match the first 5-digit of the zip code

[#2980](https://github.com/fecgov/openFEC/issues/2980#issuecomment-371876195)

- new type of filter was added : filter_multi_start_with
- this new filter is used by SA
- test case was modofied: test_filter_sa_zip

## How to test the changes locally
-run API locally and test the searching feature

### CODE REVIEW

**Comment by fec-jli** [comment]
@hcaofec good job.
Can we make this function as general function?
line 76: if kwargs.get('contributor_zip'):
change to if kwargs.get('key'):
:5 --> from arg.
Thanks

File: webservices/resources/sched_a.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def amount_column(self):
         ('contributor_id', models.ScheduleA.contributor_id),
         ('contributor_city', models.ScheduleA.contributor_city),
         ('contributor_state', models.ScheduleA.contributor_state),
-        ('contributor_zip', models.ScheduleA.contributor_zip)
+        # ('contributor_zip', models.ScheduleA.contributor_zip)
```
Comment (LindsayYoung) [inline webservices/resources/sched_a.py]: delete

**Comment by lbeaufort** [comment]
@hcaofec per our discussion in engineering sync, are you planning to add a `gin` index to the zipcode? Should this PR go back to [WIP] and no `please review` label for now?

**Comment by hcaofec** [comment]
@lbeaufort  good point. I added [wip] to the tilte. gin index has been added to dev and I am testing it. Should there a new ticket of adding index to schedule A table?

**Comment by lbeaufort** [comment]
@hcaofec awesome, thank you! 

In my mind, the `gin` index can all be in the existing PR/Issue because it's all related to the same change. @LindsayYoung thoughts?

**Pull request closed without merging**